picard/ui/logview.py

```diff
@@ -248,6 +248,8 @@ def _save_log_as_do(self):
             filter=_("Text Files (*.txt *.TXT)"),
             options=QtWidgets.QFileDialog.DontConfirmOverwrite
         )
+        if not path.endswith('.txt') and not path.endswith('.TXT'):
+            path = path + '.txt'
```

Comment:
Not necessarily wanted behaviour. Logs may also be stored with a different extension like picard.log or even without an extension.

Reply:
Although I am not entirely sure the intentions with which this feature was implemented since there is a text file filter above. @zas any thoughts?

Reply:
@samj1912 i don't know why this happens. This was a small fix. Though what i found interesting is that write operation gives output False if you havn't specifiy any extension

Reply:
Hmmm, imho we should remove `filter=_("Text Files (*.txt *.TXT)")`, this is why it fails if none of those extensions is set.

Reply:
@zas removing the filter doesn't seem to have any effect on it. Let me just check the reason behind this behaviour.

Reply:
I implemented a fix in https://github.com/metabrainz/picard/pull/840